### PR TITLE
Join multiple columns two

### LINKED ISSUE #62: Column mapping: Add lat/lon column mapping for CSV files
Latitude and longitude columns are required to be mapped for CSV data sources since there's no way to know what the user would intend otherwise.

### LINKED ISSUE #70: Column mapping: Adapt split functions in column mapping to use regex from backend, where appropriate

### LINKED ISSUE #71: Column mapping: allow for joining more than two columns

### LINKED ISSUE #75: Column-mapping: all headings should be links
- "next" button only activated after required fields are mapped (number, street)

### LINKED ISSUE #85: Column mapping: Use model data for column-mapping dropdowns
So they aren't pre-populated after a cancel.

### PR DESCRIPTION
**Pull request by meghanhade**
- Creates new route called data-format, to replace column-mapping
- Allows a field to be split, using same regex as backend to render example rows. (closes #70)
- Allows for more than two columns to be joined (closes #71)
- Lat/lon columns added for csv file types (to be made required fields in another branch) (closes #62)
- All headings are links (closes #75)
- Model data used for dropdowns (closes #85)

**issue #99 (Cancel button doesn't unload fields for column-mapping/data-format) will be handled in another branch

### CODE REVIEW

File: mirage/fixtures/responses.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export default [
```
Comment (hanbyul-here) [inline mirage/fixtures/responses.js]: Is this file used anywhere?

**Comment by hanbyul-here** [comment]
Hey the preview for this PR looks great, awesome work!  

One small thing is that when user selects already-selected one  for joining,  it doesn't let you do that without any signal. It would be good to have some warnings that you can't choose the same one, or just get rid of already selected one from dropdown would be less confusing?

**Comment by meghanhade** [comment]
@hanbyul-here ah, good point. I'll see if I can remove a previously-selected option from the drop-down.

**Comment by meghanhade** [comment]
@hanbyul-here I'm going to handle the drop-down issues in #112.

**Pull request merged**